src/runtime_src/core/edge/drm/zocl/zocl_drv.c

```diff
@@ -676,6 +677,19 @@ static int zocl_drm_platform_probe(struct platform_device *pdev)
 	if (ret)
 		goto err0;
 
+	/* Request dma channel */
+	if (ZOCL_PLATFORM_ARM64) {
+		dma_cap_zero(dma_mask);
+		dma_cap_set(DMA_MEMCPY, dma_mask);
+		zdev->zdev_dma_chan = dma_request_channel(dma_mask, 0, NULL);
+		if (!zdev->zdev_dma_chan) {
+			DRM_ERROR("Failed no DMA Channel for DMA_MEMCPY\n");
+			ret = -EBUSY;
+			goto err0;
+		}
+	}
```

Comment:
And at that time, if we don't have a DMA channel, we can simplly fail the command and fall back to memcpy

Reply:
One thing is missing is ps_ert_submit/ps_ert_query support. But we can leave them for the next PR.

Reply:
> I think this might be a little bit too aggressive. If we fail to request a DMA channel, do we have to fail to driver probe? Can we move this to the first time we see a copy bo command? This should be a one time thing and should not have impact on performance.

Agree. Using singleton instance for the dma channel and only request if when copy_bo is called first time. I don't think we need a lock here since the scheduler will loop commands one by one.